### PR TITLE
dynamic shape arg

### PR DESCRIPTION
**Pull request by apbose**
- Add name_dims to torch_tensorrt.Input — tag a dynamic axis with a name (Input(..., name_dims={0: "B"})); the same name across inputs is exported as one shared torch.export.Dim, letting users share a dim (e.g. a batch axis on input_ids and attention_mask) without a separate dynamic_shapes argument or any torch.export knowledge at the call site.
- Unblock kwarg-only nn.Module calls — relax the arg_inputs is None assertion when kwarg_inputs is given, normalize arg_inputs=None to [] (instead of [None]), and stop inputs or arg_inputs from overwriting an empty inputs tuple with None.

### CODE REVIEW

**Review by cehongwang — APPROVED** [review]
Looks ok

**Comment by apbose** [comment]
@narendasan are you suggesting something like this? B and S are the named tuples
B = shared_dims("B", min=1,  opt=4,   max=8)      # batch
S = shared_dims("S", min=16, opt=128, max=512)    # sequence

torch_tensorrt.compile(
    model, ir="dynamo",
    kwarg_inputs={
        "input_ids":      Input(shape=(B, S), dtype=torch.int64),
        "attention_mask": Input(shape=(B, S), dtype=torch.int64),
           },
)

**Comment by narendasan** [comment]
No it was more like two different ideas. 

Idea 1: Copy the export dynamic dimension system 

```py
torch_tensorrt.compile(model, 
    arg_inputs=[
        torch_tensorrt.Input({
            "min":(1,2,3,4), 
            "opt":(1,4,3,4),
            "max":(1,8,3,4),
        }, shared_dims={1:"channel"}),
        torch_tensorrt.Input({
            "min":(2,2,), 
            "opt":(4,2,),
            "max":(8,2,),
        }, shared_dims={0:"channel"}),
    ])     
```

Idea 2 (we can do later): Named dimensions 

```py
from collections import namedtuple

input_shape1 = namedtuple('input_shape', ['n', 'c', 'h', 'w'])
input_shape2 = namedtuple('input_shape', ['c', 'seq',])

torch_tensorrt.compile(model, 
    arg_inputs=[
        torch_tensorrt.Input({
            "min":input_shape1(1,2,3,4), 
            "opt":input_shape1(1,4,3,4),
            "max":input_shape1(1,8,3,4),
        }),
        torch_tensorrt.Input({
            "min":input_shape2(2,2,), 
            "opt":input_shape2(4,2,),
            "max":input_shape2(8,2,),
        }),
    ])     
```

**Review by narendasan — APPROVED** [review]
LGTM please add an example on how to use this feature then merge.
